frontend/appflowy_flutter/lib/workspace/application/home/home_setting_bloc.dart

```diff
@@ -42,6 +42,19 @@ class HomeSettingBloc extends Bloc<HomeSettingEvent, HomeSettingState> {
             var isMenuCollapsed = !state.isMenuCollapsed;
             _appearanceSettingsCubit.saveIsMenuCollapsed(isMenuCollapsed);
             emit(state.copyWith(isMenuCollapsed: isMenuCollapsed));
+            if (state.isSmallScreen) {
+              emit(state.copyWith(pinSideBar: !isMenuCollapsed));
+            } else {
+              if (state.isMenuCollapsed) {
+                emit(state.copyWith(pinSideBar: !isMenuCollapsed));
+              }
+            }
```

Comment:
```suggestion
            if (state.isSmallScreen || state.isMenuCollapsed) {
              emit(state.copyWith(pinSideBar: !isMenuCollapsed));
            }
```

Reply:
Thanks @Xazin Thanks for the suggestion. Will modify it